src/actions/conversationalSearch.js

```diff
@@ -0,0 +1,21 @@
+import { createConversationalSearchData } from '../util/create-conversational-search-data';
+
+export const IS_LOADING_CONVERSATIONAL_SEARCH = 'IS_LOADING_CONVERSATIONAL_SEARCH';
+export const CONVERSATIONAL_SEARCH_RESULT = 'CONVERSATIONAL_SEARCH_RESULT';
+
+export function fetchConversationalSearchResultStory(client, keyword) {
+  return (dispatch) => {
+    dispatch({ type: IS_LOADING_CONVERSATIONAL_SEARCH, payload: true });
+
+    client.conversationalSearch(keyword, (response) => {
+      const conversationalSearchResult = createConversationalSearchData(response);
+
+      dispatch({
+        type: CONVERSATIONAL_SEARCH_RESULT,
+        conversationalSearchResult
+      });
+
+      dispatch({ type: IS_LOADING_CONVERSATIONAL_SEARCH, payload: false });
```

Comment:
Is it redundant here?
I saw this case `CONVERSATIONAL_SEARCH_RESULT` in reducer always switch `conversationalSearchResult` to `false`

Reply:
Good catch. I was meaning to remove it from the CONVERSATIONAL_SEARCH_RESULT case. 

I prefer to have the direct action for the loading state, rather than the indirectness of CONVERSATIONAL_SEARCH_RESULT.

Reply:
Removed the part from the CONVERSATIONAL_SEARCH_RESULT case. I prefer the dedicated direct action for the loading state. Rather than the indirection via the CONVERSATIONAL_SEARCH_RESULT action. That way it becomes easier to follow what's happening in the redux dev tools as well.